igcollect/sfp_dom.py

```diff
@@ -0,0 +1,364 @@
+#!/usr/bin/env python3
+"""igcollect - SFP Digital Optical Monitoring metric collector
+
+Copyright (c) 2022 InnoGames GmbH
+
+This script reads SFP EEPROM and parses it in order to collect optical
+diagnostics data.
+
+To properly understand what happens on this script, a read of the SFF-8472
+standard is highly recommended.
+
+In some places, we gave up on making things Pythonic in detriment of shaving
+off a few ms on execution time.
+"""
+import ctypes
+import fcntl
+import socket
+import sys
+
+from argparse import ArgumentParser
+from os import listdir
+from os.path import isdir, islink, join
+from time import time
+
+IFNAMSIZ = 16
+
+SIOCETHTOOL = 0x8946
+
+ETHTOOL_GMODULEINFO = 0x00000042  # Get plug-in module information
+ETHTOOL_GMODULEEEPROM = 0x00000043  # Get plug-in module EEPROM
+
+ETH_MODULE_SFF_8079 = 0x1
+ETH_MODULE_SFF_8079_LEN = 256
+ETH_MODULE_SFF_8472 = 0x2
+ETH_MODULE_SFF_8472_LEN = 512
+ETH_MODULE_SFF_8636 = 0x3
+ETH_MODULE_SFF_8636_LEN = 256
+ETH_MODULE_SFF_8436 = 0x4
+ETH_MODULE_SFF_8436_LEN = 256
+
+SFF_A0_DOM = 92
+SFF_A0_DOM_PWRT = (1 << 3)
+SFF_A0_DOM_EXTCAL = (1 << 4)
+
+SFF_A2_BASE = 0x100
+SFF_A2_TEMP = 96
+SFF_A2_TEMP_HALRM = 0
+SFF_A2_TEMP_LALRM = 2
+SFF_A2_TEMP_HWARN = 4
+SFF_A2_TEMP_LWARN = 6
+SFF_A2_VCC = 98
+SFF_A2_VCC_HALRM = 8
+SFF_A2_VCC_LALRM = 10
+SFF_A2_VCC_HWARN = 12
+SFF_A2_VCC_LWARN = 14
+SFF_A2_BIAS = 100
+SFF_A2_BIAS_HALRM = 16
+SFF_A2_BIAS_LALRM = 18
+SFF_A2_BIAS_HWARN = 20
+SFF_A2_BIAS_LWARN = 22
+SFF_A2_TX_PWR = 102
+SFF_A2_TX_PWR_HALRM = 24
+SFF_A2_TX_PWR_LALRM = 26
+SFF_A2_TX_PWR_HWARN = 28
+SFF_A2_TX_PWR_LWARN = 30
+SFF_A2_RX_PWR = 104
+SFF_A2_RX_PWR_HALRM = 32
+SFF_A2_RX_PWR_LALRM = 34
+SFF_A2_RX_PWR_HWARN = 36
+SFF_A2_RX_PWR_LWARN = 38
```

Comment:
Please document where those values are coming from. The names of the .h files will be enough.

Reply:
Done.